src/inmanta/server/validate_filter.py

```diff
@@ -203,6 +203,42 @@ def to_query_type(self) -> Optional[Tuple[QueryType, object]]:
         return None
 
 
+class CombinedContainsFilterResourceState(Filter):
+    """Represents a valid ReleasedResourceState constraint,
+    which handles the filters as contains or not contains filters based on their values"""
+
+    # Pydantic doesn't support Generic models on python 3.6
```

Comment:
The comment was left over from the previous version, I removed it (since this one has a custom validator anyway). 
But you're right that this isn't a blocker anymore in general so I'll check if that works (and if it's simpler) and create a separate ticket for it